### PR TITLE
PR: Use CString/CStr from alloc/core (#261)

### PR DESCRIPTION
**Pull request by rzezeski**

### CODE REVIEW

File: opte/src/engine/mod.rs [hunk 1]
```diff
@@ -123,7 +124,7 @@ cfg_if! {
 
             unsafe {
                 if opte_debug != 0 {
-                    let cstr = cstr_core::CString::new(msg).unwrap();
+                    let cstr = CString::new(msg).unwrap();
```
Comment (bnaecker) [inline opte/src/engine/mod.rs:127]: Not really part of this PR, but we might want to truncate the message or something, rather than unwrapping.

File: opte/src/lib.rs [hunk 2]
```diff
@@ -242,12 +242,8 @@ impl LogProvider for KernelLog {
             LogLevel::Error => ddi::CE_WARN,
         };
 
-        unsafe {
-            ddi::cmn_err(
-                cmn_level,
-                cstr_core::CString::new(msg.to_string()).unwrap().as_ptr(),
-            )
-        }
+        let msg_arg = alloc::ffi::CString::new(msg.to_string()).unwrap();
```
Comment (bnaecker) [inline opte/src/lib.rs:245]: The extra `.to_string()` conversion here seems unnecessary, I think. `CString::new` takes anything convertible to a `Vec<u8>`, which includes `&str`.

**Review by bnaecker — APPROVED** [review]
Nice, pretty straightforward. Couple of small things, but looks good to me.

File: opte/src/engine/mod.rs (hunk 1 above)
Comment (rzezeski) [reply opte/src/engine/mod.rs:127]: IIRC, the unwrap is only in case this is invalid UTF-8, and I see no way for that to ever be the case given that all these should be strings coming from opte code or from network implementations using opte code. I.e., I can't ever see this string coming from bytes passed down by the user. But in any event, truncation does make sense in terms of line length. We really never should be making long debug/failure messages. And so we could probably replace this with something like a static 1024 byte buffer like we've talked about for the SDT probes.

File: opte/src/lib.rs (hunk 2 above)
Comment (rzezeski) [reply opte/src/lib.rs]: Done.

**Review by bnaecker — APPROVED** [review]
